api/tests/lib/seed_local_legacy_tables.py

```diff
@@ -0,0 +1,92 @@
+#
+# Populate legacy tables with mock data.
+#
+
+import datetime
+import logging
+
+import factory
+import sqlalchemy
+
+import src.adapters.db
+import src.logging
+from src.db import foreign
+from tests.src.db.models.factories import ForeignTopportunityFactory
+
+logger = logging.getLogger(__name__)
+
+
+def seed_local_source_tables() -> None:
+    with src.logging.init("seed_local_source_tables"):
+        logger.info("populating source tables with mock data")
+
+        db_client = src.adapters.db.PostgresDBClient()
+
+        with db_client.get_session() as db_session:
+            update_and_append_data(db_session)
+
+        logger.info("populating source tables done")
+
+
+def update_and_append_data(db_session):
+    max_opportunity_id = get_max_opportunity_id(db_session)
+    if max_opportunity_id:
+        update_existing_data(db_session, max_opportunity_id)
+    append_new_data(db_session, max_opportunity_id)
+    db_session.commit()
+
+
+def update_existing_data(db_session, max_opportunity_id):
+    """Update a subset of existing opportunities in the source tables."""
+
+    # Every record with id ending in 001:
+    update_ids = set(range(1, max_opportunity_id, 1000))
+    # Plus the 20 most recently created records:
+    update_ids |= set(range(max_opportunity_id - 19, max_opportunity_id + 1))
+
+    logger.info("updating rows %r" % sorted(update_ids))
+
+    for opportunity_id in update_ids:
+        factory.random.reseed_random(opportunity_id + max_opportunity_id)
+        ForeignTopportunityFactory.reset_sequence(opportunity_id, force=True)
+
+        updated_opportunity = ForeignTopportunityFactory.build()
+        updated_opportunity.pop("created_date")
+        updated_opportunity["last_upd_date"] = datetime.datetime.now()
+
+        db_session.execute(
+            sqlalchemy.update(foreign.opportunity.Topportunity)
+            .where(foreign.opportunity.Topportunity.opportunity_id == opportunity_id)
+            .values(**updated_opportunity)
+        )
+
+
+def append_new_data(db_session, max_opportunity_id):
+    count = 100000 if max_opportunity_id == 0 else 2000
+    generate_batch(db_session, max_opportunity_id + 1, count)
+
+
+def generate_batch(db_session, start_id, count):
+    """Generate a reproducible batch of opportunities in the source tables."""
+    logger.info("appending batch of %i rows starting with id %i" % (count, start_id))
+
+    factory.random.reseed_random(start_id)
+    ForeignTopportunityFactory.reset_sequence(start_id, force=True)
+
+    opportunity_rows = ForeignTopportunityFactory.build_batch(size=count)
+
+    db_session.execute(sqlalchemy.insert(foreign.opportunity.Topportunity), opportunity_rows)
+
+
+def get_max_opportunity_id(db_session):
+    max_opportunity_id = db_session.query(
+        sqlalchemy.func.max(foreign.opportunity.Topportunity.opportunity_id)
+    ).scalar()
```

Comment:
One thing I think we should look into for the factories is a way to do this automatically - they mention being able to force the sequence counter based on a startup method in https://factoryboy.readthedocs.io/en/stable/recipes.html#forcing-the-sequence-counter

Basically what you're doing here / what I did in the db-seed-local script, but in the factory itself.